tests/bluetooth/tester/src/model_handler.c

```diff
@@ -302,16 +303,24 @@ static struct bt_mesh_sensor_srv sensor_srv =
 static int sensor_data_get(struct bt_mesh_sensor_srv *srv,
 			   struct bt_mesh_sensor *sensor,
 			   struct bt_mesh_msg_ctx *ctx,
-			   struct sensor_value *rsp)
+			   sensor_value_type *rsp)
 {
 	int i;
 
 	for (i = 0; i < ARRAY_SIZE(sensors); ++i) {
 		if (sensor->type->id != sensors[i]->type->id) {
 			continue;
 		}
-
+#if !defined(CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE)
+		int err = bt_mesh_sensor_value_from_sensor_value(sensor->type->channels[0].format,
+								 &values[i], rsp);
```

Comment:
This could be its own task, but in the future the use of `sensor_value` for passing `values` should probably be changed to just passing the `raw` value. So we support everything with full precision in the tester.